pom.xml

```diff
@@ -40,7 +40,8 @@
         <maven.compiler.target>1.8</maven.compiler.target>
         <junit.jupiter.version>5.9.0</junit.jupiter.version>
         <jacoco.version>0.8.5</jacoco.version>
-        <argLine>-Dfile.encoding=UTF-8</argLine>
+	<argLine>-Dfile.encoding=UTF-8</argLine>
+	<project.build.outputTimestamp>2023-01-01T00:00:00Z</project.build.outputTimestamp>
```

Comment:
It's best to use the commit timestamp here IMO.

Reply:
What is suppose to happen when you release that property is updated.

I don’t use the release plugin but it does that.

Thus I did not put a meaningful timestamp.

In fact I would do the opposite of what you think and have a maven enforcer plugin validate that it is not the hardcoded default on release.

Reply:
Ideally, every build should be reproducible, not just releases.

Reply:
https://maven.apache.org/guides/mini/guide-reproducible-builds.html

> instead of explicitely writing a timestamp in their pom.xml, some people tend to prefer using last Git commit timestamp, like `${git.commit.time}` from [git-commit-id-maven-plugin](https://github.com/git-commit-id/git-commit-id-maven-plugin).

Sounds pretty good, right?

Reply:
The problem with `git.commit.time` is that you are relying on that tool to make the build reproducible.

That is some one should be able to take a zip or tgz of your source and build w/o git.

For example in my company we used Hg but now switched over to git. So now if we checkout a specific version from before git we would not be able to reproduce if we followed git.commit.time (unless of course you actually change the pom but I'm assuming that isn't the case).

Honestly the timestamps are not important so long as they stay the same. The only reason to have them change for release is mainly generated doc and possibly the zip aka jar time stamps.

Thus I use a some static default of the last release build timestamp for my snapshots and then change it for release (well I don't change the pom but use a properties file and a custom script to change the pom on release but I assume you do something different or use the maven versions plugin or release plugin).